cmd/init.go

```diff
@@ -0,0 +1,277 @@
+package cmd
+
+import (
+	"fmt"
+	"os"
+	"os/exec"
+	"time"
+
+	"github.com/bitswan-space/bitswan-gitops/internal/caddyapi"
+	"github.com/bitswan-space/bitswan-gitops/internal/dockercompose"
+	"github.com/bitswan-space/bitswan-gitops/internal/dockerhub"
+	"github.com/spf13/cobra"
+)
+
+type initOptions struct {
+	remoteRepo string
+	onPrem     bool
+	domain    string
+	certsDir  string
+}
+
+func defaultInitOptions() *initOptions {
+	return &initOptions{}
+}
+
+func newInitCmd() *cobra.Command {
+	o := defaultInitOptions()
+
+	cmd := &cobra.Command{
+		Use:   "init [flags] <gitops-name>",
+		Short: "Initializes a new GitOps, Caddy and Bitswan editor",
+		Args:  cobra.RangeArgs(1, 2),
+		RunE:  o.run,
+	}
+
+	cmd.Flags().StringVar(&o.remoteRepo, "remote", "", "The remote repository to clone")
+	cmd.Flags().StringVar(&o.domain, "domain", "", "The domain to use for the Caddyfile")
+	cmd.Flags().StringVar(&o.certsDir, "certs-dir", "", "The directory where the certificates are located")
+
+	return cmd
+}
+
+func (o *initOptions) run(cmd *cobra.Command, args []string) error {
+	bitswanConfig := os.Getenv("HOME") + "/.config/bitswan/"
+
+	if _, err := os.Stat(bitswanConfig); os.IsNotExist(err) {
+		if err := os.MkdirAll(bitswanConfig, 0755); err != nil {
+			return fmt.Errorf("failed to create BitSwan config directory: %w", err)
+		}
+	}
+
+	// Init shared Caddy if not exists
+	caddyConfig := bitswanConfig + "caddy"
+	if _, err := os.Stat(caddyConfig); os.IsNotExist(err) {
+		fmt.Println("Setting up Caddy...")
+		if err := os.MkdirAll(caddyConfig, 0755); err != nil {
+			return fmt.Errorf("failed to create Caddy config directory: %w", err)
+		}
+
+		// Create Caddyfile with email and modify admin listener
+		caddyfile := `
+		{
+			email info@bitswan.space
+			admin 0.0.0.0:2019
+		}`
+
+		caddyfilePath := caddyConfig + "/Caddyfile"
+		if err := os.WriteFile(caddyfilePath, []byte(caddyfile), 0644); err != nil {
+			return fmt.Errorf("failed to write Caddyfile: %w", err)
+		}
+
+
+		caddyDockerCompose, err := dockercompose.CreateCaddyDockerComposeFile(caddyConfig, o.certsDir, o.domain)
+		if err != nil {
+			return fmt.Errorf("failed to create Caddy docker-compose file: %w", err)
+		}
+
+		caddyDockerComposePath := caddyConfig + "/docker-compose.yml"
+		if err := os.WriteFile(caddyDockerComposePath, []byte(caddyDockerCompose), 0644); err != nil {
+			return fmt.Errorf("failed to write Caddy docker-compose file: %w", err)
+		}
+
+		err = os.Chdir(caddyConfig)
+		if err != nil {
+			return fmt.Errorf("failed to change directory to Caddy config: %w", err)
+		}
+
+		caddyProjectName := "bitswan-caddy"
+		caddyDockerComposeCom := exec.Command("docker-compose", "-p", caddyProjectName, "up", "-d")
+
+		fmt.Println("Starting Caddy...")
+		if err := caddyDockerComposeCom.Run(); err != nil {
+			return fmt.Errorf("failed to start Caddy: %w", err)
+		}
+
+		// wait 5s to make sure Caddy is up
+		time.Sleep(5 * time.Second)

```

Comment:
Not needed imediately but this should be done via pinging caddy every 1 seconds untill its up or like 20 secs have passed and we give up.

Reply:
Yeah I agree..created issue for that